app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+            $numVertices = count($coordinates);
+            $totalDistance = 0;
+
+            for ($i = 0; $i < $numVertices - 1; $i++) {
+                $totalDistance += $this->calculateDistance($coordinates[$i], $coordinates[$i + 1]);
+            }
+
+            for ($i = 0; $i < $numVertices - 1; $i++) {
+                $distance1 = $this->calculateDistance($coordinates[$i], $coordinates[($i + 1) % $numVertices]);
+                $distance2 = $this->calculateDistance($coordinates[($i + 1) % $numVertices], $coordinates[($i + 2) % $numVertices]);
+                $combinedDistance = $distance1 + $distance2;
+
+                if ($combinedDistance > 0.6 * $totalDistance) {
+                    // Vertex and its adjacent vertices contribute more than 25% of the total boundary path distance
+                    $spikes[] = $coordinates[($i + 1) % $numVertices];
+                }
+            }
+        }
+
+        return $spikes;
+    }
+
+    public function insertCriteriaSite($POLYGON_ID, $CRITERIA_ID, $valid)
+    {
+        $criteriaSite = new CriteriaSite();
+        $criteriaSite->polygon_id = $POLYGON_ID;
+        $criteriaSite->criteria_id = $CRITERIA_ID;
+        $criteriaSite->uuid = Str::uuid();
+        $criteriaSite->valid = $valid;
+
+        try {
+            $criteriaSite->save();
+
+            return true;
+        } catch (\Exception $e) {
+            return $e->getMessage();
+        }
+    }
+
+    public function checkBoundarySegments(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+        $geojson = DB::selectOne('SELECT ST_AsGeoJSON(geom) AS geojson FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->geojson;
+        $geojsonArray = json_decode($geojson, true);
+        $spikes = $this->detectSpikes($geojsonArray);
+        $SPIKE_CRITERIA_ID = 8;
+        $valid = count($spikes) === 0;
+        $insertionSuccess = $this->insertCriteriaSite($uuid, $SPIKE_CRITERIA_ID, $valid);
+
+        return response()->json(['spikes' => $spikes, 'geometry_id' => $uuid, 'insertion_success' => $insertionSuccess, 'valid' => $valid], 200);
+    }
+
+    public function validatePolygonSize(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $areaSqDegrees = DB::selectOne('SELECT ST_Area(geom) AS area FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->area;
+        $latitude = DB::selectOne('SELECT ST_Y(ST_Centroid(geom)) AS latitude FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->latitude;
```

Comment:
Not a big deal, but these two could be combined into a single query, which would be a little faster. Better yet would be to change these into accessors on the Eloquent model and do this instead:
```
$areaSqDegrees = $geometry->area_sq_degrees;
$latitude = $geometry->latitude;
```

Unfortunately, that route would go back to using two DB calls (although it would be easier to read). Maybe a single attribute on `PolygonGeometry` called `getDBGeometryAttribute` that calculates and returns both of these values in a single query.

Reply:
Move to getDBGeometryAttribute in model